### PR TITLE
Refactor: split jobs module into multiple files

### PR DESCRIPTION
**Pull request by MaikoTan**
Per https://github.com/quisquous/cactbot/issues/1944#issuecomment-722351242 ,
But I don't have all the max level jobs, I have only tested all Magical jobs working general, 'cause I only play casting ones.

cc @Echoring

### CODE REVIEW

**Review by quisquous — COMMENTED** [review]
Thanks for this! Overall, what a great cleanup!!  I have all my jobs at 80, so can give these all a quick test.

File: ui/jobs/jobs.js [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+  setupNin,
+  setupSam,
+  setupBrd,
+  setupMch,
+  setupDnc,
+  setupBlm,
+  setupSmn,
+  setupRdm,
+  setupBlu,
+} from './components/index.js';
```
[the remark this answers is not in the document]
Comment (MaikoTan) [reply ui/jobs/jobs.js:31]: In fact I am thinking to write a helper function in `jobs/components/index.js` accepts job name and return the corresponding component. so Bars don't need to know what are these setup function related from.

File: ui/jobs/jobs.js [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
-import { LocaleRegex } from '../../resources/translations.js';
+import ContentType from '../../resources/content_type.js';
 import Regexes from '../../resources/regexes.js';
-import NetRegexes from '../../resources/netregexes.js';
 import UserConfig from '../../resources/user_config.js';
 import { Util } from '../../resources/common.js';
 import ZoneInfo from '../../resources/zone_info.js';
 import ZoneId from '../../resources/zone_id.js';
+import { kWellFedContentTypes, kLevelMod, kMPCombatRate, kMPNormalRate, kMPUI1Rate, kMPUI2Rate, kMPUI3Rate, kMPTickInterval } from './constants.js';
+import { BuffTracker } from './BuffTracker.js';
```
[the remark this answers is not in the document]
Comment (MaikoTan) [reply ui/jobs/jobs.js:9]: Okay

File: ui/jobs/jobs.js [hunk 3]
```diff
@@ -59,570 +74,7 @@ const kPullText = {
...
-  for (let i = 0; i < classes.length; ++i)
-    div.classList.add(classes[i]);
-  element.appendChild(div);
-  const color = window.getComputedStyle(div).backgroundColor;
-  element.removeChild(div);
-  return color;
-}
-
-function makeAuraTimerIcon(name, seconds, opacity, iconWidth, iconHeight, iconText,
+export function makeAuraTimerIcon(name, seconds, opacity, iconWidth, iconHeight, iconText,
```
[the remark this answers is not in the document]
Comment (MaikoTan) [reply ui/jobs/jobs.js]: It is inside now!

File: ui/jobs/jobs.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/jobs/jobs.js:32]: Ok, sounds reasonable!

File: ui/jobs/jobs.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/jobs/jobs.js]: Sounds good!

File: ui/jobs/combo_tracker.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    else
+      this.callback(null);
+  }
+
+  AbortCombo(id) {
+    this.StateTransition(id, null);
+  }
+
+  static setup(callback) {
+    const comboTracker = new ComboTracker(kComboBreakers, callback);
```
Comment (quisquous) [inline ui/jobs/combo_tracker.js:69]: `Uncaught ReferenceError: kComboBreakers is not defined`

Comment (MaikoTan) [reply ui/jobs/combo_tracker.js:71]: Yes I just got fixed it, sorry!

**Comment by MaikoTan** [comment]
Seems like MPTicker is hard to be split out of jobs, not only should we extract a bunch of variables, but also rewrite some code so it doesn't deeply depend on jobs module.
I think those refactoring should be put in a new PR, this one can be merged now?

**Comment by quisquous** [comment]
I didn't test everything in too detailed of a way, but I switched through all the jobs and made sure all the ui showed up with no errors.  Seems good to me.  Thanks again for the refactoring!

**Pull request merged**